### PR TITLE
Configuration implementations [13026]

### PR DESCRIPTION
**Pull request by jparisu**
Over branch #63

### CODE REVIEW

File: src/cpp/configuration/implementations/DiscoveryServerConnectionAddress_configuration.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#include <ddsrouter/exceptions/UnsupportedException.hpp>
+#include <ddsrouter/types/address/Address.hpp>
+#include <ddsrouter/types/address/DiscoveryServerConnectionAddress.hpp>
+#include <ddsrouter/types/configuration_tags.hpp>
+#include <ddsrouter/types/RawConfiguration.hpp>
+
+namespace eprosima {
+namespace ddsrouter {
+
+DiscoveryServerConnectionAddress::DiscoveryServerConnectionAddress(
```
Comment (rsanchez15) [inline src/cpp/configuration/implementations/DiscoveryServerConnectionAddress_configuration.cpp:30]: If you have to import the yaml header file in all configuration files anyway, why aren't the implementations of these functions in their corresponding implementation files?
This applies to `Addresss_configuration.cpp`, `DiscoveryServerConnectionAddress_configuration.cpp`, `DomainId_configuration.cpp`, and `GuidPrefix_configuration.cpp`.

File: test/unittest/participant/participant_factory/ParticipantFactoryTest.cpp [hunk 2]
```diff
@@ -30,18 +31,51 @@ namespace eprosima {
...
+// TODO: refactor these tests so they test with type in id and with type in Configuration
+// TODO: add these functions in a common utils test
+void set_domain(
+        RawConfiguration& configuration,
+        uint16_t seed = 0)
+{
+    configuration[DOMAIN_ID_TAG] = seed;
+}
+
+ParticipantConfiguration get_random_participant_configuration(
```
Comment (rsanchez15) [inline test/unittest/participant/participant_factory/ParticipantFactoryTest.cpp:43]: I dont see anything random here and I would not randomize it as we always want to know every cases in the tests

```suggestion
ParticipantConfiguration get_participant_configuration(
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: include/ddsrouter/types/address/Address.hpp [hunk 3]
```diff
@@ -135,6 +141,22 @@ class Address
     //! Default transport protocol for address when is not set: UDP
     static TransportProtocol default_transport_protocol() noexcept;
 
+    /////
+    // YAML methods
+
+    /**
+     * @brief Construct a new Address from yaml object
+     *
+     * @param configuration : configuration where this address must be set (tags will be looked up in this same level)
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/types/address/Address.hpp:150]: :+1:

File: resources/configurations/descriptions/fields/address.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+
+#########################
```
[the remark this answers is not in the document]
Comment (jparisu) [reply resources/configurations/descriptions/fields/address.yaml:2]: Attempt in: esources/configurations/descriptions/README.md

File: include/ddsrouter/types/address/Address.hpp [hunk 5]
```diff
@@ -135,6 +141,22 @@ class Address
...
+     * @param default_transport : default transport protocol in case it is not set in yaml
+     */
+    Address(
+        const RawConfiguration& configuration,
+        TransportProtocol default_transport = Address::default_transport_protocol());
+
+    //! Dump this object in \c configuration variable (at this same level).
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/types/address/Address.hpp:157]: :+1:

File: include/ddsrouter/types/configuration_tags.hpp [hunk 6]
```diff
@@ -39,6 +39,28 @@ constexpr const char* PARTICIPANT_TYPE_TAG("type"); //! Participant Type
 // Simple RTPS related tags
 constexpr const char* DOMAIN_ID_TAG("domain"); //! Domain Id of the participant
 
+// Discovery Server related tags
+constexpr const char* LISTENING_ADDRESSES_TAG("listening-addresses"); //! TODO
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/types/configuration_tags.hpp:43]: To add comments

File: include/ddsrouter/types/endpoint/DomainId.hpp [hunk 7]
```diff
@@ -51,12 +51,23 @@ class DomainId
     DomainId (
             const DomainIdType& domain) noexcept;
 
+
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/types/endpoint/DomainId.hpp:54]: :+1:

File: src/cpp/configuration/implementations/Address_configuration.cpp [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+Address::Address(
+    const RawConfiguration& configuration,
+    TransportProtocol default_transport /*= Address::default_transport_protocol()*/)
+{
+    // Get IP
+    if (configuration[ADDRESS_IP_TAG])
+    {
+        try
+        {
+            ip_ = configuration.as<IpType>();
```
[the remark this answers is not in the document]
Comment (jparisu) [reply src/cpp/configuration/implementations/Address_configuration.cpp:38]: Life is hard

File: src/cpp/configuration/implementations/DiscoveryServerConnectionAddress_configuration.cpp (hunk 1 above)
Comment (jparisu) [reply src/cpp/configuration/implementations/DiscoveryServerConnectionAddress_configuration.cpp:30]: Already discussed, we'll leave it this way and change it in the future.

File: test/unittest/participant/participant_factory/ParticipantFactoryTest.cpp (hunk 2 above)
Comment (jparisu) [reply test/unittest/participant/participant_factory/ParticipantFactoryTest.cpp:43]: The proper name would be `arbitrary`. It does not mean to be random, but to create different random configurations depending on the seed, so we do not always use the main ones and forget about corner or complex cases.
We'll attempt this in future PRs.

**Review by rsanchez15 — APPROVED** [review]

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**